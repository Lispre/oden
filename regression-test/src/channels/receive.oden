package channels/receive/main

countDown(n) = {
  if n <= 0 then "I'm slow." else countDown(n - 1)
}

main : -> ()
main() =
  let r = go countDown(1000000)
  in {
    println("Running...")
    r <- "I'm fast!"
    println(<-r)
    println(<-r)
  }
