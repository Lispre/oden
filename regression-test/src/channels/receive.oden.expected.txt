Running...
I'm fast!
I'm slow.
